lib/hooks/queries/useAccountPoolAssetBalances.ts

```diff
@@ -16,11 +22,15 @@ export const useAccountPoolAssetBalances = (
   const query = useQuery(
     [id, accountPoolAssetBalancesRootKey, address, pool?.poolId, blockNumber],
     async () => {
-      if (isRpcSdk(sdk)) {
+      if (isRpcSdk(sdk) && pool) {
         const assets = isIndexedData(pool)
           ? pool.weights
-              .filter((weight) => weight.assetId !== "Ztg")
-              .map((weight) => JSON.parse(weight.assetId))
+              .filter((weight) => weight?.assetId !== "Ztg")
+              .map((weight) =>
+                weight?.assetId
```

Comment:
same here. Personally I'd rather use this `weight && parseAssetId(weight.assetId).unrightOr(undefined)`